packages/pluggableWidgets/app-events-native/src/AppEvents.tsx

```diff
@@ -18,61 +18,66 @@ export class AppEvents extends Component<Props> {
     private lastOnOnline = 0;
     private lastOnOffline = 0;
     private onLoadTriggered = false;
-    private timeoutHandle?: any;
+    private onTimeoutAction: ScheduledExecution | undefined;
     private unsubscribeNetworkEventListener?: NetInfoSubscription;
 
     async componentDidMount(): Promise<void> {
-        if (this.props.onResumeAction) {
-            AppState.addEventListener("change", this.onAppStateChangeHandler);
-        }
-
-        if (this.props.onTimeoutAction) {
-            const schedule = this.props.timerType === "once" ? setTimeout : setInterval;
-            this.timeoutHandle = schedule(() => executeAction(this.props.onTimeoutAction), this.props.delayTime * 1000);
-        }
-
-        if (this.props.onOnlineAction || this.props.onOfflineAction) {
-            this.isConnected = (await NetInfo.fetch()).isConnected;
-            this.unsubscribeNetworkEventListener = NetInfo.addEventListener(this.onConnectionChangeHandler);
-        }
-
-        if (!this.onLoadTriggered && this.props.onLoadAction?.canExecute) {
-            this.onLoadTriggered = true;
-            executeAction(this.props.onLoadAction);
-        }
+        AppState.addEventListener("change", this.onAppStateChangeHandler);
+        this.registerScheduledExecution();
+        await this.registerNetworkEvents();
+        this.triggerOnLoadEvent();
     }
 
     componentWillUnmount(): void {
-        if (this.props.onUnloadAction && this.props.onUnloadAction.canExecute) {
-            executeAction(this.props.onUnloadAction);
-        }
+        AppState.removeEventListener("change", this.onAppStateChangeHandler);
+        this.onTimeoutAction?.cancel();
+        this.unRegisterNetworkEvents();
+        // TODO: Suggestion Deprecate this behavior.
+        // Keeping it as most likely someone is misusing this for tracking navigation events.
+        this.triggerOnUnloadEvent();
+    }
 
+    render(): null {
+        return null;
+    }
+
+    private async onResume(): Promise<void> {
         if (this.props.onResumeAction) {
-            AppState.removeEventListener("change", this.onAppStateChangeHandler);
+            executeAction(this.props.onResumeAction);
         }
+        this.onTimeoutAction?.schedule();
+        this.lastOnResume = Date.now();
+        // TODO: Not clear if onResume should wait for registerNetworkEvents. This can take a few seconds while app has resumed
```

Comment:
I think it should because from users' perspective, resuming is bit more complete (everything is ready and hooked up) than what we have in technical terms. 

Does that mean line 51 will be before 45 ?

Reply:
Yes sure but that event can take seconds depending on device

Reply:
So if you tell client's they should use this for onResume; I as a user of your app might have resumed; navigated and the client still hasn't triggered onResume

Reply:
Again this has to do with expectations. I want my onResume call to happen as soon as my page loaded; not in any random interval from the time it loaded till this async resolves. With as it is right now that will happen; so onResume will always be very unreliable.